spec/models/user_access_spec.rb

```diff
@@ -528,20 +238,609 @@
+    let!(:facility_2) { create(:facility, facility_group: facility_group_2) }
+    let!(:facility_3) { create(:facility, facility_group: facility_group_3_1) }
+    let!(:facility_4) { create(:facility, facility_group: facility_group_3_2) }
+    let!(:facility_5) { create(:facility) }
+    let!(:facility_6) { create(:facility) }
+
+    let!(:user_1) { create(:user, :with_phone_number_authentication, registration_facility: facility_1) }
+    let!(:user_2) { create(:user, :with_phone_number_authentication, registration_facility: facility_2) }
+    let!(:user_3) { create(:user, :with_phone_number_authentication, registration_facility: facility_3) }
+    let!(:user_4) { create(:user, :with_phone_number_authentication, registration_facility: facility_4) }
+    let!(:user_5) { create(:user, :with_phone_number_authentication, registration_facility: facility_5) }
+
+    let!(:admin_1) { create(:admin, :call_center, organization: organization_1) }
+    let!(:admin_2) { create(:admin, :call_center, organization: organization_1) }
+    let!(:admin_3) { create(:admin, :call_center, organization: organization_3) }
+    let!(:admin_4) { create(:admin, :call_center, organization: organization_3) }
+    let!(:admin_5) { create(:admin, :call_center) }
+
+    let!(:manager) { create(:admin, :manager) }
+    let!(:viewer_all) { create(:admin, :viewer_all) }
+    let!(:viewer_reports_only) { create(:admin, :viewer_reports_only) }
+    let!(:call_center) { create(:admin, :call_center) }
+
+    context "non power users" do
+      context "#accessible_organizations" do
+        let!(:permission_matrix) {
+          [
+            [manager, :manage, organization_3, [organization_3]],
+            [manager, :view_pii, organization_3, [organization_3]],
+            [manager, :view_reports, organization_3, [organization_3]],
+            [manager, :manage_overdue_list, organization_3, [organization_3]],
+
+            [viewer_all, :manage, organization_3, []],
+            [viewer_all, :view_pii, organization_3, [organization_3]],
+            [viewer_all, :view_reports, organization_3, [organization_3]],
+            [viewer_all, :manage_overdue_list, organization_3, [organization_3]],
+
+            [viewer_reports_only, :manage, organization_3, []],
+            [viewer_reports_only, :view_pii, organization_3, []],
+            [viewer_reports_only, :view_reports, organization_3, [organization_3]],
+            [viewer_reports_only, :manage_overdue_list, organization_3, []],
+
+            [call_center, :manage, organization_3, []],
+            [call_center, :view_pii, organization_3, []],
+            [call_center, :view_reports, organization_3, []],
+            [call_center, :manage_overdue_list, organization_3, [organization_3]]
+          ]
+        }
+
+        it "returns the organizations an admin can perform actions on" do
+          permission_matrix.each do |admin, action, current_resource, accessible_resources|
```

Comment:
Suggested rename for readability here for this and all the other matrices below: `accessible_resources` -> `expected_resources`